### ISSUE TITLE
[APP SUBMITTED]: SEARCHQUEUE-MANUAL-270408 :: [BLUETIGERS] :: Deluge: Unable to send Torrent: Return code undefined [4423fe9]

### ISSUE THREAD

**Issue by thenamelessthing** [SickRage/sickrage-issues#374]
### INFO
Python Version: **2.7.6 (default, Jun 22 2015, 17:58:13) [GCC 4.8.2]**
Operating System: **Linux-3.13.0-74-generic-x86_64-with-Ubuntu-14.04-trusty**
Locale: UTF-8
Branch: **master**
Commit: SickRage/SickRage@4423fe95002315b812b4ef9867ea6df7d94c2083
Link to Log: https://gist.github.com/32ed18ab8d046cfd50f3
### ERROR
```
SEARCHQUEUE-MANUAL-270408 :: [BLUETIGERS] :: Deluge: Unable to send Torrent: Return code undefined [4423fe9]
```
---
_STAFF NOTIFIED_: @SickRage/owners @SickRage/moderators

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [x] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by thenamelessthing** [comment]
### INFO
Python Version: **2.7.6 (default, Jun 22 2015, 17:58:13) [GCC 4.8.2]**
Operating System: **Linux-3.13.0-74-generic-x86_64-with-Ubuntu-14.04-trusty**
Locale: UTF-8
Branch: **master**
Commit: SickRage/SickRage@4423fe95002315b812b4ef9867ea6df7d94c2083
Link to Log: https://gist.github.com/811e5fa9d429cf539a58
### ERROR
```
SEARCHQUEUE-DAILY-SEARCH :: Deluge: Unable to send Torrent: Return code undefined [4423fe9]
```
---
_STAFF NOTIFIED_: @SickRage/owners @SickRage/moderators

**Comment by thenamelessthing** [comment]
Here is more details: 

OS: Ubuntu 14.04.3
Version:	Branch: master
Commit: 4423fe95002315b812b4ef9867ea6df7d94c2083
Version: v5.0.3
User:	--------
Locale:	Language: en_US
Encoding: UTF-8
Configuration File:	/opt/sickrage/config.ini
Database:	/opt/sickrage/sickbeard.db
Cache Directory:	/opt/sickrage/cache
Log Directory:	/opt/sickrage/Logs
Arguments:	['-q', '--daemon', '--nolaunch', '--pidfile=/var/run/sickrage/sickrage.pid', '--datadir=/opt/sickrage/']
Python Version:	2.7.6 (default, Jun 22 2015, 17:58:13) [GCC 4.8.2]


I don't know why I got this error, no special action are done, I use Deluge with Deluge-webui with self-signed ssl.

**Comment by thenamelessthing** [comment]
It's a new error message since the last SR update (I think)

**Comment by thenamelessthing** [comment]
I have these warning too...  
2015-12-31 11:32:44 SEARCHQUEUE-DAILY-SEARCH :: Connection error to getURL https://www.bluetigers.ca/download.php?id=143813&hit=1 Error: u"error HTTPSConnectionPool(host='www.bluetigers.ca', port=443): Max retries exceeded with url: /download.php?id=143813&hit=1 (Caused by ConnectTimeoutError(, 'Connection to www.bluetigers.ca timed out. (connect timeout=30)'))" [4423fe9]
2015-12-31 11:32:44 SEARCHQUEUE-DAILY-SEARCH :: Torrent file content is empty [4423fe9]

**Comment by thenamelessthing** [comment]
It look SR is unable to download the torrent file properly? (empty)

**Comment by neoatomic** [comment]
@thenamelessthing 
There where some update to some torrent providers, but not to bleutigers if i recall right.

The error basically tells that sickrage cant make a connection to bluetigers.ca.
I just tried the link that failed, but it seems to beworking fine and downloads the torrent. So i get the feeling this is a SSL related issue. (not sickrage related)
I would try to update Python to version 2.7.10 or 2.7.11. and then try again.
Versions below that need decencies for SSL, so if those are not installed correctly you get SSL probs...
And for testing you can always revert to an earlier commit-version of Sickrage. That way you can exclude that the course is a recent update.
If you find its indeed a bug, then please let me know.

And a Happy new year. :)

**Comment by thenamelessthing** [comment]
I have try with python 2.7.11 without any change. So how can I revert to an earlier commit?

**Comment by thenamelessthing** [comment]
After googling I have found this command to revert temporary to an earlier commit:

git checkout f6f2afb5df9ec69b9d6a3a839912bfd761520428 (I have revert avery commit up to the 23dec and always the same problem)  So the problem still persist, how can we fix it?

**Comment by neoatomic** [comment]
Ok, that at least excludes any recent changes that are coursing the error for you.
Just to be sure, bluetigers.ca is not blocked in your country right.?

The error basically tells that SickRage was unable to connect to the www.bluetigers.ca site, and probbebly then sends a empty torrent to Deluge.
As the site uses HTTPS, the first thing i would check if SSL is installed/setup and working correctly.
However you could try the steps below. Could be that the cache has gone bad.

Try this. : 
* Shutdown Sickrage.
* Remove the cache.db fine. 
* Remove cache/Mako & cache/sessions folders if they exist.
* Start Sickrage.

@miigotu Might be worth checking if we can catch empty/failed torrents being sent to the download client.That would also prevent they getting stuck on `snatched`

**Comment by labrys** [comment]
@thenamelessthing Please wrap your logs in \`\`\`\ as it's much easier to read that way.  I have done it for you this time.

**Comment by thenamelessthing** [comment]
I have try to remove the cache.db and the sessions folder (no trace for the Mako).  I have find that the torrent seem to be sent to Deluge but incorrectly because Deluge doesn't set the label and the right on complete saved folder.  It use the Deluge default folder and not the one specified by SR.

**Comment by thenamelessthing** [comment]
https://sc-cdn.scaleengine.net/i/fc7102aa034bfe4d3a5509743eaf6fcb.png

**Comment by fernandog** [comment]
so deluge is raising an error for unicode erros while adding torrent?
Strange that your locale is UTF-8

**Comment by duramato** [comment]
```Operating System: Linux-3.13.0-74-generic-x86_64-with-Ubuntu-14.04-trusty```
Ubuntu 14.04 has a locale problem ,best solution is updating to 15

**Comment by thenamelessthing** [comment]
I have fixed the problem with this solution:

I have to investigate it deeper and this solution helps me to solve it:

dpkg-reconfigure locales
choose en_CA.UTF-8 UTF-8 (default) and/or en_US.UTF-8 UTF-8

apt-get install console-data
dpkg-reconfigure console-data
set pc / qwerty / french / Standard / Standard or any other on your choice

dpkg-reconfigure console-setup
set Encoding to use on the console: UTF-8 set Charecter set: Latin1 & Latin 5 - , vga, 16 or any other on your choice

in file /etc/default/locale set

LANG=en_CA.UTF-8
LANGUAGE=en_CA.UTF-8

**Comment by fernandog** [comment]
@thenamelessthing can you help us with something?

we need the HTML for BLUETIGERS when doing a search. we need parse size, seeders and leechers from torrent. so we can filter torrents without min seeds and have size for failed downloads.

do a simple search like 'les revenats S01E01' and paste the html (using chrome viewsource for example)

**Comment by thenamelessthing** [comment]
@fernandog, the site is down for maintenance until 11. So I'll do that next week.

**Comment by fernandog** [comment]
ok! NP!

@duramato

**Comment by fernandog** [comment]
@thenamelessthing are they online again?

**Comment by thenamelessthing** [comment]
@fernandog I will check that during the evening. I'm at work at the moment..

**Comment by thenamelessthing** [comment]
@fernandog I have looked if it's up, but not yet. It will be back online around 17h (- 5 EST).

**Comment by miigotu** [comment]
looks up from here =¶

**Comment by thenamelessthing** [comment]
Login not possible at this time. Only the chat as guest.

**Comment by thenamelessthing** [comment]
Still in mainteance, on the chat they mention it will be online around 12h PM the 12/01  (-5 est)

**Comment by fernandog** [comment]
@thenamelessthing can you ask them if would be an API ?

**Comment by thenamelessthing** [comment]
@fernandog I sent a message to an admin about the API.  I also retrieved the source page you request.

[source page.txt](https://github.com/SickRage/sickrage-issues/files/87893/source.page.txt)

**Comment by fernandog** [comment]
@miigotu is that ^^ good?

**Comment by thenamelessthing** [comment]
I got the answer about the API. No API,  :-(

**Comment by miigotu** [comment]
Also will need the download link for one of those torrents in that page.

**Comment by thenamelessthing** [comment]
@miigotu I don't know if i'm doing something wrong but to get the .torrent I have to click on the result to go on the detailed page. On the detailled page I have the url: http://www.bluetigers.ca/download.php?id=134404&name=Les.Revenants.S02E07.FRENCH.HDTV.XviD-ASPHiXiAS.torrent

**Comment by miigotu** [comment]
yes, I just needed the url from the details page, because I can use the name and id from index to build that url ^

**Comment by thenamelessthing** [comment]
Do you need anything else? ;-)

**Comment by miigotu** [comment]
Nope, should be enough to work on it. I will need it tested in the next day or so though.

**Comment by fernandog** [comment]
https://github.com/SickRage/sickrage-issues/issues/725

**Comment by thenamelessthing** [comment]
If you need help on this please let me know.

**Comment by duramato** [comment]
@adaur do you have an account?  Can you take a look at the size seeders and leechers?

**Comment by jgrossrieder** [comment]
I got this error sometime, but not related to bluetigers:

> 2016-12-15 14:13:45 WARNING SEARCHQUEUE-BACKLOG-72108 :: Deluge: Unable to send Torrent
> 2016-12-15 14:13:45 INFO SEARCHQUEUE-BACKLOG-72108 :: Downloading XXX.YYY.ZZZ from T411
